### ISSUE TITLE
Script not detecting sidestake

### ISSUE THREAD

**Issue by ghost** [makeasnek/FindTheMag2#21]
When running with `SCRIPTED_RUN:bool=True` the script prints that it will work for dev on Linux. On MacOS it refuses to start due to that it can't find the sidestake.

**Comment by makeasnek** [comment]
Can you please provide output from the script and verify you are on the latest version of FTM? Thank you for the bug report! A copy of your gridcoin wallet conf file would also be helpful, you can e-mail to makeasnek{at}gmail.com

**Comment by sweede-se** [comment]
```
************************************************************************************************************************************************************************************************
*                                                                                       FINDTHEMAG V2.0                                                                                        *
************************************************************************************************************************************************************************************************
*                              | HOURSOFF | TOTALCREDIT | ATIME | ACTIME | ACPT | TASKS | WTIME | CPUTIME | CREDIT/HR | R-WTIME | WEIGHT | GRC/HR | GRC/DAY | USD/HR R/P |
* asteroids                    |-0.02     |412.71       |0.43   |0.28    |31.75 |13.0   |5.53   |3.66     |74.57      |5.53     |987.0   |0.079   |1.884    |0.0010/-0.00|
* amicable numbers             |0.01      |61525.74     |2.34   |4.48    |6836.1|9.0    |21.08  |40.29    |2919.28    |21.08    |1       |0.056   |1.338    |0.0007/-0.01|
* odlk1                        |0.0       |42.36        |0.17   |0.08    |7.06  |6.0    |1.04   |0.49     |40.58      |1.04     |1       |0.044   |1.062    |0.0005/-0.01|
* nfs                          |0.01      |3376.0       |0.42   |0.25    |48.93 |69.0   |28.94  |17.58    |116.64     |28.94    |1       |0.034   |0.810    |0.0004/-0.01|
* einstein                     |0.0       |0.0          |4.11   |2.74    |0.0   |2.0    |8.23   |5.48     |0.0        |8.23     |1       |0.000   |0.000    |0           |
* tn-grid                      |0.0       |193.01       |0.87   |0.43    |64.34 |3.0    |2.62   |1.29     |73.8       |2.62     |NA      |0.000   |0.000    |0           |
* rosetta                      |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* milkyway                     |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* numberfields                 |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* srbase                       |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* universe                     |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* gpugrid.net                  |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* yoyo                         |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* sidock                       |0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
* worldcommunitygrid.org/boinc/|0.0       |0.0          |0.0    |0.0     |0.0   |0.0    |0.0    |0.0      |0.0        |0.0      |1       |0.000   |0.000    |0           |
************************************************************************************************************************************************************************************************
*Sleep reason: NONE                                         *Info:                                                      GRC Price: 0.012*
When you started using this tool, your average mag/hr was: 0.0000 now it is 0.1559
Hours crunched for you vs dev: 34.5|0.0
Consider donating to this app's development directly or via sidestake: RzUgcntbFm8PeSJpauk6a44qbtu92dpw3K. Sidestaking means you can skip crunching for dev
Use Ctrl+C to exit FTM and return BOINC to previous config
************************************************************************************************************************************************************************************************
```

FindTheMag2 downloaded from github two days ago, so I guess it´s the latest version.

Sending config files by e-mail.

**Comment by HeyMerlin** [comment]
@sweede-se 
I'm assuming that the output is from the Linux run. What about the output of the failed run on Mac?

On the Mac is the last message you see in the printed or in the log file:
`Sidestaking must be setup for BOINC control on OS X as "crunch for dev" is not an option. Re-run the script to set this up.`

Does the Mac fail whether or not `SCRIPTED_RUN:bool=` is set true or false?


@makeasnek 
If I'm parsing the code correctly, in `__main__`:
It looks like the exit condition triggered on the Mac is here:
```
    if "DARWIN" in found_platform.upper() and not check_sidestake_results:
        print_and_log(
            'Sidestaking must be setup for BOINC control on OS X as "crunch for dev" is not an option. Re-run the script to set this up.',
            "ERROR",
        )
        quit()
```
When this code is encountered and `SCRIPTED_RUN:bool=True`, then `check_sidestake_results` is always `False`, as shown below:

`check_sidestake_results` is set here:
```
    # Get project list from Gridcoin wallet and/or gridcoinstats
    check_sidestake_results = False
    foundation_address = "bc3NA8e8E3EoTL1qhRmeprbjWcmuoZ26A2"
    developer_address = "RzUgcntbFm8PeSJpauk6a44qbtu92dpw3K"
```
Then checking the sidestake is skipped due to `SRIPTED_RUN` being True:
```
    else:
        if not SCRIPTED_RUN:
            # Check sidestakes
```

The question I have is... why is checking for sidestaking skipped in a scripted run?

**Comment by sweede-se** [comment]
@HeyMerlin It's from my Windows PC.

The Mac is only failing to detect side stake when boolean are set to true.

```If you'd like to say thank you to the developers of this tool, please help us buy our next round of energy drinks by sending GRC to:
RzUgcntbFm8PeSJpauk6a44qbtu92dpw3K
Starting control of BOINC...
Sidestaking must be setup for BOINC control on OS X as "crunch for dev" is not an option. Re-run the script to set this up.```

**Comment by HeyMerlin** [comment]
Thanks @sweede-se. That confirms my thinking of why this is happening. 

I'll have to defer to @makeasnek  to confirm and recommend a solution. Meanwhile I'll keep looking at the code to figure out the logic of not checking for the sidestaking.

**Comment by makeasnek** [comment]
Sorry for the delay on this, I will patch this soon. @HeyMerlin is correct as to why this is happening.

**Comment by makeasnek** [comment]
Fixed in latest release v 3.0
